webpack/assets/javascripts/react_app/components/common/Slot/Slot.test.js

```diff
@@ -0,0 +1,185 @@
+      <React.Fragment>
+        <Fill
+          overrideProps={{ text: 'This is given by a prop' }}
+          slotId="slot-3"
+          id="some-key-1"
+          weight={100}
+        />
+
+        <Slot id="slot-3" multi={false}>
+          <SlotComponent />
+        </Slot>
+      </React.Fragment>
+    );
+
+    wrapper.update();
+    expect(toJson(wrapper.find('Slot'))).toMatchSnapshot();
+  });
+  it('default slot', () => {
+    const wrapper = integrationTestHelper.mount(
+      <Slot id="slot-4" multi>
+        <SlotComponent text="Default Value" />
+      </Slot>
+    );
+
+    wrapper.update();
+    expect(toJson(wrapper.find('Slot'))).toMatchSnapshot();
+  });
+  it('multiple slot with override props', () => {
+    const wrapper = integrationTestHelper.mount(
+      <React.Fragment>
+        <Fill
+          overrideProps={{ text: 'This should be the second', key: 1 }}
+          slotId="slot-5"
+          id="some-key-1"
+          weight={100}
+        />
+        <Fill
+          overrideProps={{ text: 'This should be the first', key: 2 }}
+          slotId="slot-5"
+          id="some-key-2"
+          weight={200}
+        />
+        <Slot id="slot-5" multi>
+          <SlotComponent text="Default Value" />
+        </Slot>
+      </React.Fragment>
+    );
+    wrapper.update();
+    expect(toJson(wrapper.find('Slot'))).toMatchSnapshot();
+  });
+
+  it('slot with multi override props should take max weight', () => {
+    const wrapper = integrationTestHelper.mount(
+      <React.Fragment>
+        <Fill
+          overrideProps={{ text: 'This should not be in the snap' }}
+          slotId="slot-6"
+          id="some-key-1"
+          weight={100}
+        />
+        <Fill
+          overrideProps={{ text: 'This text should be in the snap' }}
+          slotId="slot-6"
+          id="some-key-2"
+          weight={200}
+        />
+        <Slot id="slot-6">
+          <SlotComponent text="Default Value" />
+        </Slot>
+      </React.Fragment>
+    );
+    wrapper.update();
+    expect(toJson(wrapper.find('Slot'))).toMatchSnapshot();
+  });
+
+  it('multi slot with override props fill amd component fill', () => {
+    const wrapper = integrationTestHelper.mount(
+      <React.Fragment>
+        <Fill
+          overrideProps={{ key: 'def', text: 'This should be in the snap' }}
+          slotId="slot-10"
+          id="some-key-1"
+          weight={100}
+        />
+        <Fill slotId="slot-10" id="some-key-2" weight={100}>
+          <div key="abc"> Also this should be in the snap </div>
+        </Fill>
+        <Slot id="slot-10" multi>
+          <SlotComponent text="Default Value" />
+        </Slot>
+      </React.Fragment>
+    );
+    wrapper.update();
+    expect(toJson(wrapper.find('Slot'))).toMatchSnapshot();
+  });
+
+  it('check redux lifecycle when rendering fills on a slot', () => {
+    integrationTestHelper.takeStoreSnapshot();
+    integrationTestHelper.takeActionsSnapshot();
```

Comment:
What benefit do those snapshots give?
I think the point is to use them inside the other tests so together with your `jsx` snapshots you have a snapshot of the action and the state that cause your `jsx` snapshot.

Reply:
Look at the snapshot that this has created, basically it got all of the actions and corresponding reducers, so with this we test the entire functionality of redux

Reply:
Yes but it creates snapshots in this test but they based on the logic ran in the other tests :/
Therefore I believe each test should use his own `IntegrationTestHelper` instance and keep its own state snapshots (that related to the context of the test).